### PR TITLE
Patch & Ragdoll & AnnounceScpDeath

### PR DESCRIPTION
**Pull request by warquys**
Patch the interface bug when published of the SynapseInjector. The interface become a class when it was published.
Add roleID and RoleType on the Ragdoll and add oldRoleID on the player. Add a new method AnnounceScpDeath on Map for annonce the Death of a SCP with a raison.

### CODE REVIEW

**Comment by AlmightyLks** [comment]
Mind proving the fix for the interface / class bug?

**Comment by warquys** [comment]
no problem
I post cella on discord because it will be easier

**Comment by GrafDimenzio** [comment]
The change in the Loader so that the interfaces are actually interfaces is a good addition, but the other things aren't completely thought through

**Comment by warquys** [comment]
I was fix it tomorrow

**Comment by AlmightyLks** [comment]
> ScpRecon_**f**_Type

**Comment by warquys** [comment]
> ScpRecon_f_Type ?

File: Synapse/Api/Enum/ScpReconfType.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Synapse.Api.Enum
+{
+    public enum ScpReconfType
+    {
+        Unspecified = 0,
+        ByMtf = 1,
+        ByChi = 2,
+        ByRsc = 3,
+        ByCdp = 4,
```
Comment (AlmightyLks) [inline Synapse/Api/Enum/ScpReconfType.cs:9]: Would prefer the "By" to be removed

File: Synapse/Api/Enum/ScpReconfType.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        Tesla = 5,
+        Nuke = 6,
+        Decontamination = 7,
+        ByUnknown = 8
```
Comment (AlmightyLks) [inline Synapse/Api/Enum/ScpReconfType.cs:13]: Read above

File: Synapse/Api/Enum/ScpReconfType.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Synapse.Api.Enum
+{
+    public enum ScpReconfType
```
Comment (AlmightyLks) [inline Synapse/Api/Enum/ScpReconfType.cs:3]: `ScpReconfType` -> `ScpRecontainmentType`
1. Typo
2. Don't be shy to type out stuff, "Recon" is a bit eh

File: Synapse/Api/Map.cs [hunk 4]
```diff
@@ -105,6 +106,43 @@ public void AnnounceScpDeath(string scp)
...
+                    cause = $". SCP {scp} SUCCESSFULLY TERMINATED . BY CLASSD PERSONNEL";
+                    break;
+                case ScpReconfType.ByUnknown:
+                    cause = $". SCP {scp} SUCCESSFULLY TERMINATED . CONTAINMENTUNIT UNKNOWN";
+                    break;
+                case ScpReconfType.Unspecified:
+                default:
+                    cause = $". SCP {scp} SUCCESSFULLY TERMINATED . TERMINATION CAUSE UNSPECIFIED";
+                    break;
+            }
```
Comment (AlmightyLks) [inline Synapse/Api/Map.cs:142]: You may want to consider using a switch-assignment, see https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#switch-expressions

File: Synapse/Api/Ragdoll.cs [hunk 5]
```diff
@@ -25,6 +26,11 @@ public Ragdoll(RoleType roletype, Vector3 pos, Quaternion rot, Vector3 velocity,
 
         public string Name => ragdoll.name;
 
+        public RoleType RoleType
+        {
+            get => Server.Get.Host.ClassManager.Classes.FirstOrDefault(role => role.classColor == ragdoll.owner.ClassColor).roleId;
```
Comment (AlmightyLks) [inline Synapse/Api/Ragdoll.cs:31]: How does the color of a class define what roletype a ragdoll is?
Also, isn't the classcolor the one from spectator, or whatever they are at the point _after_ dropping a ragdoll?

File: Synapse/SynapseExtensions.cs [hunk 6]
```diff
@@ -97,6 +97,21 @@ public static SynapseItem GetSynapseItem(this ItemPickupBase pickupbase)
         return item;
     }
 
+    public static ScpReconfType GetScpReconfType(this PlayerStats.HitInfo info)
```
Comment (AlmightyLks) [inline Synapse/SynapseExtensions.cs:100]: Same naming change applies

**Review by AlmightyLks — CHANGES_REQUESTED** [review]

File: Synapse/Api/Enum/ScpReconfType.cs (hunk 1 above)
Comment (AlmightyLks) [reply Synapse/Api/Enum/ScpReconfType.cs:9]: @GrafDimenzio , opinion?

Comment (warquys) [reply Synapse/Api/Enum/ScpReconfType.cs:9]: I think you're right

**Comment by warquys** [comment]
I made the changes you advised me

**Pull request merged**